config/components/prometheus/monitor_tls_patch.yaml

```diff
@@ -0,0 +1,34 @@
+# Patch for Prometheus ServiceMonitor to enable secure TLS configuration
+# using certificates managed by cert-manager
+#
+# [PROMETHEUS WITH CERTMANAGER]
+# ${METRICS_PROMETHEUS_CLIENT_CERT_SECRET_NAME} should be replaced with a prometheus client secret.
+# The same certificate for the server ("metrics-server-cert") can potentially be also used for the client.
+# It is preferable to use prometheus client certificate if available:
+# Either by creating a secret in the jobset controller namespace
+# or by using prometheus mounted secret which can be configured via certFile and keyFile in the service_monitor.yaml
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: controller-manager-metrics-monitor
+  namespace: system
+spec:
+  endpoints:
+    - path: /metrics
+      port: https
+      scheme: https
+      bearerTokenFile: /var/run/secrets/kubernetes.io/serviceaccount/token
+      tlsConfig:
+        insecureSkipVerify: false
```

Comment:
I think, this should be `insecureSkipVerify: true` and there will be a "op: replace" command to replace with these values when cert-manager is enabled, in order to be compatible with https://github.com/kubernetes-sigs/jobset/pull/863/files#diff-89204119394a9cad711f1fdf90f9dd71be4f7ae8ef5bca4cd545661002a419bdR37-R55

Reply:
This is applied only when used in `PROMETHEUS WITH CERTMANAGER` configuration. 

We do use `insecureSkipVerify: true` otherwise in https://github.com/kubernetes-sigs/jobset/blob/789481a770af28fd94bf7724c54bd11d85759aea/config/components/prometheus/monitor.yaml#L23